src/main/java/school/hei/haapi/endpoint/rest/mapper/RetakeExamMapper.java

```diff
@@ -33,21 +34,24 @@ public school.hei.haapi.model.RetakeExam toDomainCrupDate(CrupdateRetakeExam cru
         .student(studentUser)
         .session(retakeExamSession)
         .course(course)
+        .registrationDate(crupdateRetakeExam.getRegistrationDate())
         .build();
   }
 
   public RetakeExam toRest(school.hei.haapi.model.RetakeExam retakeExam) {
     return new RetakeExam()
         .id(retakeExam.getId())
         .course(courseMapper.toRest(retakeExam.getCourse()))
-        .session(retakeExamSessionMapper.toRest(retakeExam.getSession()));
+        .session(retakeExamSessionMapper.toRest(retakeExam.getSession()))
+        .registrationDate(retakeExam.getRegistrationDate());
   }
 
   public school.hei.haapi.model.RetakeExam toDomain(RetakeExam retakeExam) {
     return school.hei.haapi.model.RetakeExam.builder()
         .id(retakeExam.getId())
         .course(courseMapper.toDomain(retakeExam.getCourse()))
         .session(retakeExamSessionMapper.toDomain(retakeExam.getSession()))
+        .registrationDate(Instant.now())
```

Comment:
Can't you make it only through database ? 

So its value will be really null before persistence and current timestamp value after the persistence ? In that way, we really can consider that it was generated only after persistence. So here, just really map retakeExam.getRegistrationDate